### PR TITLE
refactor: uptake and merge card

### PR DESCRIPTION
**Pull request by emilyjablonski**
# Pull Request Template

## Issue Overview

This PR addresses #3904

- [x] This change addresses only certain aspects of the issue

## Description

This PR includes a refactor to the Card components that ended up being required to do the passwordless toggle on the sign-in page cleanly. 

It (1) fixes some bugs that were introduced in mandated accounts, (2) abstracts the AccountCard component into a BloomCard component, and (3) uptakes the BloomCard in the application as well so that we can have consistent padding on our form cards throughout the public site. 

It also uptakes BloomCard on the two remaining account screens that still had the UIC FormCard - forgot password and reset password.

## How Can This Be Tested/Reviewed?

Known changes are [here](https://docs.google.com/document/d/1jaqWzAQUkD11th_dodXnPXi8SFGS4-iDoo0-XNZf1Pc/edit?usp=sharing)

And there should be no regressions on:
* The create account page
* The account edit page
* The my applications page
* The account dashboard

I'd recommend testing on the [deploy preview](https://deploy-preview-3922--bloom-exygy-dev.netlify.app/)

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs (this is just a stepping stone)
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

**Comment by ColinBuyck** [comment]
Too small to even call it a nit but could we align the button to field spacing between the sign-in page and the forgot password page to 24px
Screenshot 2024-03-11 at 11 56 14 AM
Screenshot 2024-03-11 at 11 56 38 AM

File: shared-helpers/src/types/scss.d.ts
Comment (ColinBuyck) [inline shared-helpers/src/types/scss.d.ts:1]: What does this do?

**Comment by ColinBuyck** [comment]
Thoughts on aligning these with the other alert widths?
Screenshot 2024-03-11 at 12 18 44 PM

File: shared-helpers/src/types/scss.d.ts
Comment (emilyjablonski) [reply shared-helpers/src/types/scss.d.ts:1]: It's just something for vscode to stop throwing an error when we import scss files

**Comment by emilyjablonski** [comment]
Fixed the above, and also these two requests from design:
* Revert the listing title / progress nav card on application pages to have 32px horizontal padding and the same font sizes as existed before
* Vertical padding on mobile cards is now 24px

**Comment by ColinBuyck** [comment]
Could we turn the mandated accounts flag on for the deploy previews? @emilyjablonski

**Comment by emilyjablonski** [comment]
@ColinBuyck We can't because mandated accounts won't be on in core, but if needed we could spin up a new Netlify environment / heroku backend, or can test locally w yarn build

**Review by ColinBuyck — APPROVED** [review]
Ah okay @emilyjablonski! Well then this LGTM 🐁  once all convos are resolved with design. Really appreciate the clear patterns and how clean everything looks now!!

**Comment by emilyjablonski** [comment]
Tysm for the thorough review!

**Pull request merged**